:recycle: Refactor avatar list with new api

xr1s opened the pull request:
## 描述 / Description

使用 simnet 的 `get_genshin_character_detail` 重构了一下 `/avatars` 命令

xr1s committed:
:recycle: Refactor avatar_list

xr1s committed:
:art: Organize avatar_list, assets_service imports

xr1s committed:
:recycle: AssetsService re-define ICON_TYPE

omg-xtao commented on plugins/genshin/avatar_list.py:
```diff
@@ -36,141 +38,131 @@
+            # 不能直接判断效果文案中是否存在「替代冲刺」，因为流浪者的元素战技效果文案中也有「替代冲刺」
+            if skill.skill_type == 1 and skill.id not in [10013, 10413]
+        ]
+        for skill in skills:  # 返回的技能等级是已经被命座效果强化过的，为了统一处理，这里将增益去除（等级 -3）
```
应该添加额外字段判断是否满级，展示等级应仍然显示获得增益之后的等级

xr1s replied:
额外字段我加一下。

获得增益后展示等级 +3 的逻辑在 jinja 里，我没改。
这里是接口之间的区别，之前的 get_character_detail 返回没有 +3，这个 get_genshin_character_detail 接口 +3 了，所以我需要额外 -3.

omg-xtao commented on plugins/genshin/avatar_list.py:
```diff
@@ -230,38 +202,47 @@ async def avatar_list(self, update: "Update", _: "ContextTypes.DEFAULT_TYPE"):
+        assert message is not None
+        assert message.text is not None
+
+        show_all = "全部" in message.text or "all" in message.text  # 是否发送全部角色
```
```suggestion
        show_all = "全部" in message.text or "all" in message.text  # 是否发送全部角色
```

luoshuijs replied:
估计是 message 返回的类型为 `Meesage | None` 导致的要求需要检测 message 是否为 None
因为 handler 的问题 这里的 message 一定是不为 None 的
只能 `message : "Message" = update.effective_message`

omg-xtao commented on plugins/genshin/avatar_list.py:
```diff
@@ -230,38 +202,47 @@ async def avatar_list(self, update: "Update", _: "ContextTypes.DEFAULT_TYPE"):
+        await message.reply_chat_action(ChatAction.TYPING)
+
+        # 获取角色和武器详情数据
+        async with self.helper.genshin(user_id, player_id=uid, offset=offset) as client:
```
```suggestion
        # 获取角色和武器详情数据
        async with self.helper.genshin(user_id, player_id=uid, offset=offset) as client:
            notice = await message.reply_text(f"{config.notice.bot_name}需要收集整理数据，还请耐心等待哦~")
            self.add_delete_message_job(notice, delay=60)
            await message.reply_chat_action(ChatAction.TYPING)
```

omg-xtao commented on plugins/genshin/avatar_list.py:
```diff
@@ -230,38 +202,47 @@ async def avatar_list(self, update: "Update", _: "ContextTypes.DEFAULT_TYPE"):
+            if filter_weapon_types:
+                filter_weapon_type_names = {weapon_types.name for weapon_types in filter_weapon_types}
+                characters = [c for c in characters if c.weapon.type in filter_weapon_type_names]
+            avatar_datas: list[AvatarData] = await self.get_avatar_datas(characters, client)
```
过滤后 characters 可能为空

omg-xtao commented on plugins/genshin/avatar_list.py:
```diff
@@ -274,35 +255,3 @@ async def avatar_list(self, update: "Update", _: "ContextTypes.DEFAULT_TYPE"):
             "[bold]练度统计[/bold]发送图片成功",
             extra={"markup": True},
         )
-
```
支持 inline 使用练度统计

luoshuijs commented on resources/genshin/avatar_list/main.jinja2:
```diff
@@ -105,15 +105,15 @@
                     {% endif %}
 
-                    {% if skill.max_level == skill.level %}
+                    {% if skill.level == 10 %}
```
我在想如果这里 `leave` 出现 13 会怎么样

luoshuijs replied:
实在不行可以考虑 `{% if skill.level >= 10 %}`

xr1s replied:
我上面处理过，对于有命座增益的技能特地 -3 了，不会有 13
不过我还是把 max_level 加上吧，先硬编码到代码里，以防以后真的出现满级等级 > 10 的

xr1s committed:
Apply suggestions from code review

Co-authored-by: omg-xtao <100690902+omg-xtao@users.noreply.github.com>

xr1s committed:
:bug: Several fixes

1. Add max_level in SkillData for flexibility;
2. Early exit when no result character after filter;
3. Put back inline message handler.

omg-xtao committed:
improve
